### ISSUE TITLE
afl-showmap slow with long harness initialization time

### ISSUE THREAD

**Issue by dloffre-snl** [AFLplusplus/AFLplusplus#1086]
I have a test harness with substantial initialization time and fast fuzz time (i.e. 10 seconds to init, >500 execs/sec after that).  `afl-fuzz` works as expected, only paying the startup cost at initialization time.  

I've noticed that `afl-showmap` doesn't behave this way on my harness.  Instead, `afl-showmap` will pay the initialization cost for every input in the directory.  Is there a fix for this, or a technical reason why `afl-showmap` behaves different from `afl-fuzz` in this respect?

The behavior can be verified with this dummy harness:

```cc
// test_fuzzer.cc
//
// Test fuzz harness with long init delay
// 
// Compilation:
//    afl-clang-fast -o test_fuzzer test_fuzzer.cc afl-compiler-rt.o utils/aflpp_driver/aflpp_driver.o
// 
// Running with showmap:
//    AFL_PRINT_FILENAMES=1 afl-showmap -C -t 20000 -i IN -o /dev/null -- test_showmap/test_fuzzer

#include <stddef.h>
#include <stdint.h>
#include <unistd.h>
#include <stdio.h>

extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
  int flags[100];
  if (size > 0 && data[0] == 'H')
    if (size > 1 && data[1] == 'I')
       if (size > 2 && data[2] == '!')
       __builtin_trap();

  return 0;
}


extern "C" int LLVMFuzzerInitialize(int *argc, char ***argv) {
  printf("LLVMFuzzerInitialize got called!\n");
  sleep(10);
  return 0;
}
```

**Comment by vanhauser-thc** [comment]
how are you calling afl-showmap?

**Comment by domenukk** [comment]
Looking through the code again, the forkserver should indeed behave exactly the same in afl-showmap and afl-fuzz

**Comment by vanhauser-thc** [comment]
@ Dominik - My guess he is using it wrong or with the wrong expectation, that’s why I want to know the command line :)

**Comment by dloffre-snl** [comment]
The afl-showmap command line is commented in the code:

```
AFL_PRINT_FILENAMES=1 afl-showmap -C -t 20000 -i IN -o /dev/null -- test_showmap/test_fuzzer
```

In this case `IN` is a directory with ~10 files and each file takes 10 seconds.

**Comment by vanhauser-thc** [comment]
@dloffre-snl  you are correct there is an issue, the behaviour makes no sense ....

**Comment by vanhauser-thc** [comment]
fixed in f760e80
